apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4ClaimResourceProvider.java

```diff
@@ -0,0 +1,105 @@
+package gov.cms.bfd.server.war.r4.providers;
+
+import ca.uhn.fhir.model.primitive.IdDt;
+import ca.uhn.fhir.rest.annotation.IdParam;
+import ca.uhn.fhir.rest.annotation.Read;
+import ca.uhn.fhir.rest.api.server.RequestDetails;
+import ca.uhn.fhir.rest.server.IResourceProvider;
+import ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException;
+import com.codahale.metrics.MetricRegistry;
+import com.newrelic.api.agent.Trace;
+import gov.cms.bfd.server.war.commons.LoadedFilterManager;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.Claim;
+import org.hl7.fhir.r4.model.IdType;
+import org.springframework.stereotype.Component;
+
+/** This FHIR {@link IResourceProvider} adds support for R4 {@link Claim} resources. */
+@Component
+public final class R4ClaimResourceProvider implements IResourceProvider {
+  /**
+   * A {@link Pattern} that will match the {@link Claim#getId()}s used in this application, e.g.
+   * <code>f-1234</code> or <code>m--1234</code> (for negative IDs).
+   */
+  private static final Pattern CLAIM_ID_PATTERN = Pattern.compile("(\\p{Alpha}+)-(-?\\p{Alnum}+)");
```

Comment:
any reason to not just limit this to `f` and `m` prefixes?
```
Pattern.compile("([fm])-(-?\\p{Alnum}+)")
```

Reply:
I was just copying the pattern used in EoB.  I guess you could make the argument that it would increase the required code changes if adding another claim type, or that it doesn't allow for as specific of an error when determining what the issue was, but really I was just following the logic of the existing one.

Reply:
Whether we use single character or longer prefixes it could be helpful if the regex specifically matched only our valid prefixes rather than `Alpha`.